File: samples/java/07-Simple-Native-Functions/src/main/java/plugins/mathplugin/Math.java
// Copyright (c) Microsoft. All rights reserved.

package plugins.mathplugin;

import com.microsoft.semantickernel.skilldefinition.annotations.DefineSKFunction;
import com.microsoft.semantickernel.skilldefinition.annotations.SKFunctionInputAttribute;

public class Math {

    public Math() {}
    @DefineSKFunction(name = "Sqrt", description = "Take the square root of a number")
    public String sqrt(@SKFunctionInputAttribute(description = "The number to take the square root of") String input) {
        return Double.toString(java.lang.Math.sqrt(Double.parseDouble(input)));
    }
    
}

File: samples/dotnet/07-Simple-Native-Functions/plugins/MathPlugin/Math.cs
// Copyright (c) Microsoft. All rights reserved.

using System.ComponentModel;
using System.Globalization;
using Microsoft.SemanticKernel;

namespace Plugins.MathPlugin;

public class Math
{
    [SKFunction, Description("Take the square root of a number")]
    public double Sqrt([Description("The number to take a square root of")] double input)
    {
        return System.Math.Sqrt(input);
    }
}
